src/Symfony/Component/Translation/LocaleSwitcher.php

```diff
@@ -11,23 +11,31 @@
 
 namespace Symfony\Component\Translation;
 
+use Symfony\Component\Routing\RequestContext;
 use Symfony\Contracts\Translation\LocaleAwareInterface;
 
 /**
  * @author Kevin Bond <kevinbond@gmail.com>
  */
-final class LocaleSwitcher implements LocaleAwareInterface
+class LocaleSwitcher implements LocaleAwareInterface
 {
+    private string $defaultLocale;
+
     /**
      * @param LocaleAwareInterface[] $localeAwareServices
      */
-    public function __construct(private string $locale, private iterable $localeAwareServices)
-    {
+    public function __construct(
+        private string $locale,
+        private iterable $localeAwareServices,
+        private ?RequestContext $requestContext = null,
```

Comment:
This has been mentioned on https://github.com/symfony/symfony/pull/45793#discussion_r830604119
Adding the request context here makes everything simpler (no need for the compiler pass, and no need for `LocaleAwareRequestContext`, which looks like pure boilerplate to me.)

Reply:
Maybe I missed it but I don't think `RequestContext` is wired.

Reply:
oops, fixed (with tests)